contracts/JoepegAuctionHouse.sol

```diff
@@ -0,0 +1,1154 @@
+        }
+
+        // 3. Transfer final amount (post-fees) to seller
+        {
+            _currency.safeTransferFrom(_from, _to, finalSellerAmount);
+        }
+    }
+
+    /// @notice Transfer fees and funds in AVAX using WAVAX to royalty recipient,
+    /// protocol, and seller
+    /// @param _collection address of ERC721 token
+    /// @param _tokenId token id of ERC721 token
+    /// @param _to seller's recipient
+    /// @param _amount amount of WAVAX being transferred
+    /// @param _minPercentageToAsk minimum percentage of the gross amount that goes to ask
+    function _transferFeesAndFundsWithWAVAX(
+        address _collection,
+        uint256 _tokenId,
+        address _to,
+        uint256 _amount,
+        uint256 _minPercentageToAsk
+    ) internal {
+        // Initialize the final amount that is transferred to seller
+        uint256 finalSellerAmount = _amount;
+
+        // 1. Protocol fee
+        {
+            uint256 protocolFeeAmount = _calculateProtocolFee(
+                _collection,
+                _amount
+            );
+            address _protocolFeeRecipient = protocolFeeRecipient;
+
+            // Check if the protocol fee is different than 0 for this strategy
+            if (
+                (_protocolFeeRecipient != address(0)) &&
+                (protocolFeeAmount != 0)
+            ) {
+                IERC20(WAVAX).safeTransfer(
+                    protocolFeeRecipient,
+                    protocolFeeAmount
+                );
+                finalSellerAmount -= protocolFeeAmount;
+            }
+        }
+
+        // 2. Royalty fee
+        {
+            (
+                address royaltyFeeRecipient,
+                uint256 royaltyFeeAmount
+            ) = royaltyFeeManager.calculateRoyaltyFeeAndGetRecipient(
+                    _collection,
+                    _tokenId,
+                    _amount
+                );
+
+            // Check if there is a royalty fee and that it is different to 0
+            if (
+                (royaltyFeeRecipient != address(0)) && (royaltyFeeAmount != 0)
+            ) {
+                IERC20(WAVAX).safeTransfer(
+                    royaltyFeeRecipient,
+                    royaltyFeeAmount
+                );
+                finalSellerAmount -= royaltyFeeAmount;
+
+                emit RoyaltyPayment(
+                    _collection,
+                    _tokenId,
+                    royaltyFeeRecipient,
+                    address(WAVAX),
+                    royaltyFeeAmount
+                );
+            }
+        }
+
+        // Ensure seller gets minimum expected fees
+        if (
+            finalSellerAmount * PERCENTAGE_PRECISION <
+            _minPercentageToAsk * _amount
+        ) {
+            revert JoepegAuctionHouse__FeesHigherThanExpected();
+        }
+
+        // 3. Transfer final amount (post-fees) to seller
+        {
+            IERC20(WAVAX).safeTransfer(_to, finalSellerAmount);
+        }
+    }
```

Comment:
These 2 functions are now identical if you use WAVAX as the currency in the `_transferFeesAndFunds` function. The WAVAX one should be removed to simplify the contract